src/base/ddl_log.c

```diff
@@ -917,7 +917,7 @@ logddl_write_end_for_csql_fileinput (const char *fmt, ...)
 {
   FILE *fp = NULL;
   char buf[DDL_LOG_BUFFER_SIZE] = { 0 };
-  int len = 0;
+  unsigned int len = 0;
```

Comment:
len stores the result of snprintf and snprintf returns a size_t.
```suggestion
  size_t len = 0;
```
As a result of len being unsigned, conditions comparing `len < 0` should be removed too.